synthesizer/process/src/stack/call/mod.rs

```diff
@@ -382,6 +382,11 @@ impl<N: Network> CallTrait<N> for Call<N> {
             // Inject the existing circuit.
             A::inject_r1cs(r1cs);
 
+            // If in 'CheckDeployment' mode, set the expected constraint limit.
+            if let CallStack::CheckDeployment(_, _, _, constraint_limit) = &registers.call_stack() {
+                A::set_constraint_limit(*constraint_limit);
+            }
+
```

Comment:
I'm not convinced this logic is necessary.

When triggering `call`, no logic above this resets the constraint limit AFAIK.